packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -431,13 +431,17 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
     if (velocity != 0.0 || ((oldPosition.minScrollExtent == newPosition.minScrollExtent) && (oldPosition.maxScrollExtent == newPosition.maxScrollExtent))) {
       return super.adjustPositionForNewDimensions(oldPosition: oldPosition, newPosition: newPosition, isScrolling: isScrolling, velocity: velocity);
     }
-    if (oldPosition.pixels < oldPosition.minScrollExtent) {
-      final double oldDelta = oldPosition.minScrollExtent - oldPosition.pixels;
-      return newPosition.minScrollExtent - oldDelta;
-    }
-    if (oldPosition.pixels > oldPosition.maxScrollExtent) {
-      final double oldDelta = oldPosition.pixels - oldPosition.maxScrollExtent;
-      return newPosition.maxScrollExtent + oldDelta;
+    // If resize from zero, recover position from cache, so don't try to
+    // maintain the relative overscroll.
+    if (oldPosition.minScrollExtent != 0.0 || oldPosition.maxScrollExtent != 0.0) {
```

Comment:
@Hixie Hi, Please review here, in this case, the _PagePosition.applyViewportDimension will correct the pixels due to resizing from zero, (such as oldPosithon is (pixels: 800, minScrollExtent : 0.0, maxScrollExtent: 0.0)) and it will cause `RangeMaintainingScrollPhysics.adjustPositionForNewDimensions` mistake maintain the relative overscroll.
In addition, if the #63146 reland, we can remove this check and it works well.

Reply:
I'll try to land the old PR again to make this moot.

Reply:
(but with && -> ||)